features/google/src/model/importers/activities-importer-html.js

```diff
@@ -12,15 +12,97 @@ class ActivityHtmlParser {
     _extractDate(text) {
         // Ignoring the timestamp for now - we don't need hourly accuracy at the
         // time of writing this.
-        const datePattern =
-            /([A-Z][a-z][a-z]) ([0-9]{1,2}), ([0-9]{4,4}), ([0-9:]{7,8} [A|P]M) ([A-Z]{3,4})/;
-        const match = text.match(datePattern);
-        const month = match?.[1];
-        const day = match?.[2];
-        const year = match?.[3];
-
-        const dateString = `${day} ${month} ${year}`;
-        return dateString ? new Date(dateString) : null;
+
+        // Known formats:
+        // 3 Feb 2001, 16:05:06 UTC
+        // Feb 3, 2001, 4:05:06 PM UTC
+        // 03.02.2001, 16:05:06 UTC
+
+        let date = new Date(text);
+
+        if (date && date !== null && date.toString() !== "Invalid Date") {
+            return date;
+        }
+
+        date = null;
+
+        const datePatterns = [
+            /([A-Z][a-z][a-z]) ([0-9]{1,2}), ([0-9]{4,4}), ([0-9:]{7,8} [A|P]M) ([A-Z]{3,4})/,
+            /([0-9]{1,2}) ([A-Z][a-z][a-z]) ([0-9]{4,4}), ([0-9:]{8,8}) ([A-Z]{3,4})/,
+            /([0-9]{1,2}).([0-9]{1,2}).([0-9]{4,4}), ([0-9:]{8,8}) ([A-Z]{3,4})/,
+        ];
+
+        // Add +1 to the index because the entire matched string will be at index 0 in match.
+        const dateMaps = [
+            {
+                day: 2,
+                month: 1,
+                year: 3,
+                time: 4,
+                timezone: 5,
+            },
+            {
+                day: 1,
+                month: 2,
+                year: 3,
+                time: 4,
+                timezone: 5,
+            },
+            {
+                day: 1,
+                month: 2,
+                year: 3,
+                time: 4,
+                timezone: 5,
+            },
+        ];
```

Comment:
if you wanna do it this way, maybe using directly a `map` here makes it easier to read it. 
so the key is the `pattern` and the value the order that you expect (the `dateMap`). 
As now the second and third object of `dateMaps` are the same...

Reply:
In my experience it can get messy with map pretty fast. I prefer to keep them like this.